### PR TITLE
adding logic for South African phone numbers and Mobile numbers

### PR DESCRIPTION
**Pull request by toyerma**
Added some logic for valid South African phone numbers and mobile numbers

### CODE REVIEW

File: chance.js [hunk 1]
```diff
@@ -1339,6 +1340,30 @@
...
+            case 'za':
+                if (!options.mobile) {
+                    numPick = this.pick([
+                       '01' + this.pick(['0', '1', '2', '3', '4', '5', '6', '7', '8']) + self.string({ pool: '0123456789', length: 7}),
+                       '02' + this.pick(['1', '2', '3', '4', '7', '8']) + self.string({ pool: '0123456789', length: 7}),
+                       '03' + this.pick(['1', '2', '3', '5', '6', '9']) + self.string({ pool: '0123456789', length: 7}),
+                       '04' + this.pick(['1', '2', '3', '4', '5','6','7', '8','9']) + self.string({ pool: '0123456789', length: 7}),   
+                       '05' + this.pick(['1', '3', '4', '6', '7', '8']) + self.string({ pool: '0123456789', length: 7}),
+                    ]);
+                    phone = options.formatted ? numPick : numPick;
```
Comment (GrayedFox) [inline chance.js]: This line is redundant no? If options.formatted is defined, numPick, else, numPick?

File: chance.js [hunk 2]
```diff
@@ -1339,6 +1340,30 @@
...
+                } else {
+                    numPick = this.pick([
+                        '060' + this.pick(['3','4','5','6','7','8','9']) + self.string({ pool: '0123456789', length: 6}),
+                        '061' + this.pick(['0','1','2','3','4','5','8']) + self.string({ pool: '0123456789', length: 6}),
+                        '06'  + self.string({ pool: '0123456789', length: 7}),
+                        '071' + this.pick(['0','1','2','3','4','5','6','7','8','9']) + self.string({ pool: '0123456789', length: 6}),
+                        '07'  + this.pick(['2','3','4','6','7','8','9']) + self.string({ pool: '0123456789', length: 7}),
+                        '08'  + this.pick(['0','1','2','3','4','5']) + self.string({ pool: '0123456789', length: 7}),                     
+                    ]);
+                    phone = options.formatted ? (numPick) : (numPick);
```
Comment (GrayedFox) [inline chance.js]: Same as first comment (redundant ternary operator, unless I'm mistaken)

Comment (GrayedFox) [inline chance.js]: you could short circuit here with phone = `options.formatted || numPick`

Comment (toyerma) [inline chance.js]: Yes, that could work too, thanks for that. I have you tested the || code?

Comment (GrayedFox) [inline chance.js]: I haven't! I could do so a bit later, little busy at the moment building stuff for work but if you're interested in testing locally you could log `phone` on the next line and test different inputs. Sorry for the random strangers code review, was just looking over this library to see if it's a good fit for us :)

Comment (toyerma) [inline chance.js]: @GrayedFox tested and works. Thank you

**Pull request merged**